client.go

```diff
@@ -390,6 +392,32 @@ type Client struct {
 // Initialize a default *Client instance
 var DefaultClient = newClient(nil)
 
+func (c *Client) SetIgnoreErrors(errs []string) error {
+	newRegexps := make([]*regexp.Regexp, 0, len(errs))
+	var err error
+	for _, errStr := range errs {
+		var r *regexp.Regexp
+		if r, err = regexp.Compile(errStr); err != nil {
+			return fmt.Errorf("failed to compile regexp for %q: %v", errStr, err)
+		}
+		newRegexps = append(newRegexps, r)
+	}
+	c.ignoreErrors = append(c.ignoreErrors, newRegexps...)
+	return nil
+}
+
+func (c *Client) shouldExcludeErr(errStr string) bool {
+	for _, r := range c.ignoreErrors {
+		if r.MatchString(strings.ToLower(errStr)) {
```

Comment:
I don't agree that we should blindly `ToLower` here since this can be set on the regular expression itself by prepending `(?i)` to your pattern.

Also, for performance, we should handle this the same way that we do in `raven-js` and compile the list of `errs []string` into one single `regexp` object instead of iterating and evaluating each of them linearly. If they're compiled into one big regexp, it can be evaluated in `O(log N)` time instead of `O(n)` as we're doing here.